services/QuillLMS/client/app/bundles/Staff/components/comprehension/activities.tsx

```diff
@@ -1,19 +1,33 @@
 import * as React from "react";
 import { Link } from 'react-router-dom';
 import { useQuery } from 'react-query';
-
 import Navigation from './navigation'
-
 import { ActivityInterface } from '../../interfaces/comprehensionInterfaces';
 import { fetchActivities } from '../../utils/comprehension/activityAPIs';
 import { DataTable, Error, Spinner } from '../../../Shared/index';
+import { fetchAppSetting } from "../../../Shared/utils/appSettingAPIs";
 
 const Activities = ({ location, match }) => {
 
   // cache activity data for updates
-  const { data } = useQuery("activities", fetchActivities);
+  const { data: activitiesData } = useQuery("activities", fetchActivities);
+
+  const appSettingName = 'foo'
+  const appSettingQueryKey = `appSetting-${appSettingName}`
+  const { data: appSettingData } = useQuery({
+    queryKey: [appSettingQueryKey, appSettingName],
+    queryFn: fetchAppSetting
+  });
+
+
+  let appSettingValue = false;
+  if (appSettingData && appSettingData.appSetting && appSettingData.appSetting[appSettingName]) {
+    appSettingValue = appSettingData.appSetting[appSettingName]
+  } 
+  console.log("appSettingValue: ", appSettingValue)
```

Comment:
Not planning to deploy this file's changes. This is an example integration.

Reply:
We'll probably end up pulling it out into a custom hook like this:

```
const [appSetting, setAppSetting] = React.useState<string>(null);

handleAppSettingUpdate({ appSetting, appSettingName, setAppSetting, appSettingData });
```

and in the custom hook file:
``` 
function handleAppSettingUpdate({ appSetting, appSettingName, setAppSetting, appSettingData }) {
  return React.useEffect(() => {
    if (!appSetting && appSettingData && appSettingData.appSetting && appSettingData.appSetting[appSettingName]) {
      const appSettingValue = appSettingData.appSetting[appSettingName];
      setAppSetting(appSettingValue);
    } 
  }, [appSettingData]);
}
```

Reply:
So, as you mention, it would be really great if we could abstract more of this into the parent class so that we could could just have `hasFeatureX` be set boilerplate code.. It might be worth working with @eadams17 or @emilia-friedberg a bit to think of a way to do this.

```javascript
// Similar to Eric's but returning a boolean and abstracting more
const [hasAppSetting, setHasAppSetting] = React.useState<boolean>(false);
handleHasAppSetting({hasAppSetting, setHasAppSetting, 'feature_key'})

Reply:
Thanks for the idea - the integration API is down to 2 lines now.